[gekhro] iP

gekhro opened the pull request:
### Dukepro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's, 
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you're Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

gekhro committed:
Level-1

gekhro committed:
level 2

gekhro committed:
level 3

gekhro committed:
level 4

Commit:
level5

Commit:
adds delete function

Commit:
branch-Level-7

Commit:
Level-8

Commit:
Merge branch 'branch-Level-7'

kayyenl commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+                    continue;
+                }
+
+                else if (input.equals("mark")){
```
Hey! I know you haven't gotten to the code-standard refinement stage yet, but I thought I'd just leave some comments on your code style (its a fast fix, don't worry).

kayyenl replied:
for your conditionals and everything that requires a () bracket, if there is curly braces after it, it would be best to leave a white space before the curly brace. eg.

```
else if (condition) {
    code...
}
```

kayyenl commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,35 @@
+        this.isDone = true;
+    }
+
+    public void markAsNotDone(){
```
I realize you have done this in several other code files, so I guess you could just follow that.

kayyenl commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+import javax.swing.text.DateFormatter;
+
+public class Deadline extends Task{
+    private LocalDate by;
```
Although this code is just a WIP, I would just like to leave a comment: perhaps changing out the private LocalDate "by" with dateBy, or a more descriptive name would be better! I realized that you use isDone and is... for your boolean variables, so that's good, and your other variables should also reflect that level of precise descriptiveness (in camelcase, of course.)

kayyenl replied:
Thanks so much for reading! Hope the (somewhat wordy) comments has been somewhat useful. 😄

Commit:
A-MoreOOP + A-Packages

Commit:
change filepath for testing

Commit:
added gradle but no support

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into HEAD

WeideTeo commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,57 @@
+    public Ui () {
+    }
+
+    public void showWelcome() {
```
Good usage of camel case !!

WeideTeo commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+    private LocalDate due_by;
+    private String output;
+
+    public Deadline(String description, String due_by) throws DukeException{
```
You might wanna leave a space after the method name

WeideTeo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+        switch(commandinput) {
+
+            case "TODO":
+                return new AddTodo(input);
```
Good usage of simple expressions rather than complicated one

WeideTeo reviewed (commented):
Overall, excellent job of the code style. Great !!

Commit:
Level-9

Commit:
branch-A-JavaDoc

Commit:
code contained a variable which did not follow java coding standard

edited the variable name to follow the java coding standard

Commit:
Merge branch 'Level-9' into A-CodingStandard

Commit:
Merge branch 'branch-A-JavaDoc' into A-CodingStandard

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat
#	src/main/java/duke/commands/Delete.java
#	src/main/java/duke/commands/Mark.java
#	src/main/java/duke/commands/Unmark.java
#	src/main/java/duke/tasks/Deadline.java

Commit:
Implemented gradle and resolved merge conflicts

Removed files containing different gradle.build causing error
-utilities/build.gradle
-list/build.gradle
-app/build.gradle

Commit:
Branch: Add JUnit Tests

Commit:
Branch: Add Jar file

changed build.gradle to add javafx dependencies

Commit:
Merge tag 'A-JUnit'

Commit:
Update: CheckSyle
implemented checkstyle into gradle build to check for violations

Commit:
Update: Adds GUI for duke, removes textbased ui for duke

Commit:
Merge branch 'Level-10'

Commit:
Fix: Checkstyle errors

gekhro committed:
Add: A-Assertions

added code to assert that the index will be within range for delete, find and list commands

gekhro committed:
Fix: ExceptionHandling

fixed exceptionhandling logic in adding deadline, event and todo tasks

gekhro committed:
Add: Enums

gekhro committed:
Merge pull request #2 from gekhro/A-Assertions

Add: A-Assertions

Commit:
Update: more assertions

Commit:
Revert "Update: more assertions"

This reverts commit 97fcc3a0692c3c79a77f4c5db9b55a39dd9eb9a1.

Commit:
Merge remote-tracking branch 'origin/master'

gekhro committed:
Merge branch 'master' into heads/A-CodeQuality

gekhro committed:
Merge pull request #3 from gekhro/heads/A-CodeQuality

Heads/a code quality

Commit:
Add: reminder command

Commit:
Add: userguide

Commit:
A-MoreErrorHandling

Commit:
Edit: more java doc comments

Commit:
Fix: jar file
